Sourcery/Parsing/Parser.swift

```diff
@@ -374,21 +376,14 @@ final class Parser {
                         parameter.argumentLabel = argumentLabels[index]
                     }
 
-                    if let actualTypeName = typeName(for: parameter.unwrappedTypeName, containingType: type, typealiases: typealiases) {
-                        parameter.type = unique[actualTypeName]
-                    } else {
-                        parameter.type = unique[parameter.unwrappedTypeName]
-                    }
+                    parameter.type = resolveType(parameter.unwrappedTypeName, type, typealiases)
                 }
 
-                if method.returnTypeName != "Void" {
-                    if let actualTypeName = typeName(for: method.unwrappedReturnTypeName, containingType: type, typealiases: typealiases) {
-                        method.returnType = unique[actualTypeName]
-                    } else {
-                        method.returnType = unique[method.unwrappedReturnTypeName]
-                    }
+                if method.returnTypeName.name != "Void" {
```

Comment:
I just remembered that we should also check for `()`, can you add this check here too? Or maybe better will be to incapsulate that into a property on `TypeName` like `isVoid` and use it here?